dashboard-app/src/dashboard/dataManipulation/aggregatedToGraphData.ts

```diff
@@ -0,0 +1,40 @@
+import { AggregatedData } from './logsToAggregatedData';
+import { toUnitDuration } from './util';
+import { omit, map, Dictionary } from 'ramda';
+import { DurationUnitEnum } from '../../types';
+import { GraphColourList } from '../../styles/design_system';
+
+export const STACKED_TABLE_OPTIONS = {
+  legend: {
+    position: 'right',
+  },
+  scales: {
+    xAxes: [{
+      stacked: true,
+    }],
+    yAxes: [{
+      stacked: true,
+    }],
+  },
+};
+
+export const aggregatedToStackedGraph = (aggData: AggregatedData, unit: DurationUnitEnum) => {
+  const [groupBy, ...labels] = aggData.headers;
+  return {
+    labels,
+    datasets: aggData.rows.map((x, i) => {
+      const label = x[groupBy];
+      const withoutGroupBy = omit([groupBy], x);
+      const dataObject = map((v) =>
+        typeof v === 'object'
+          ? toUnitDuration(unit, v)
+          : v
+        , withoutGroupBy) as Dictionary<any>;
```

Comment:
No need for this to have an `any`. We know the type is `Dictionary<string | number>`. If you want better type inference try out `Objects.mapValues` from `twine-util`?

Reply:
Would be useful to chat about how much we wanna use ramda with ts. I already pulled this out of a pipe to remove type errors.

Reply:
🤷🏼‍♂️ use it when it's useful, don't use it when it isn't. If you want to trial that type library that supposed to give better inference then go for it.